packages/user-banning-nodejs/src/plugin.ts

```diff
@@ -0,0 +1,197 @@
+          }
+
+          return null;
+        },
+      },
+      {
+        path: HANDLE_BASE_PATH + '/ban',
+        method: 'get',
+        verifySessionOptions: {
+          sessionRequired: true,
+          overrideGlobalClaimValidators: (globalClaimValidators, session, userContext) => {
+            return [...globalClaimValidators, PermissionClaim.validators.includes('ban-user')];
+          },
+        },
+        handler: async (req, res, _, userContext) => {
+          // make sure the user is allowed to get the ban status
+          // if (!(await isAuthorised(req, res, "admin"))) {
+          //   sendUnauthorisedAccess(res);
+          //   return null;
+          // }
+
+          let tenantId = await req.getKeyValueFromQuery('tenantId');
+          if (!tenantId) {
+            res.setStatusCode(400);
+            res.sendJSONResponse({
+              status: 'BAD_INPUT_ERROR',
+              message: 'tenantId is required',
+            });
+            return null;
+          }
+
+          // make sure the request is valid
+          let userId: string | undefined = await req.getKeyValueFromQuery('userId');
+          let email: string | undefined = await req.getKeyValueFromQuery('email');
+
+          if (email) {
+            const user = await SuperTokens.listUsersByAccountInfo(tenantId, {
+              email: email.toLowerCase(),
+            });
+            userId = user?.[0]?.id;
+          }
+
+          userId = userId?.trim();
+
+          if (!userId) {
+            res.setStatusCode(400);
+            res.sendJSONResponse({
+              status: 'BAD_INPUT_ERROR',
+              message: 'userId or email is required',
+            });
+            return null;
+          }
+
+          // get the ban status
+          const result = await userBanningService.getBanStatus(userId, userContext);
+
+          if (result.status === 'OK') {
+            res.setStatusCode(200);
+            res.sendJSONResponse(result);
+          } else {
+            res.setStatusCode(400);
+            res.sendJSONResponse(result);
+          }
+
+          return null;
+        },
+      },
+    ],
+    overrideMap: {
+      session: {
+        functions: (originalImplementation) => {
+          return {
+            ...originalImplementation,
+
+            createNewSession: async (input) => {
+              const result = await userBanningService.getBanStatus(input.userId, input.userContext);
+              if (result.status !== 'OK') {
+                throw new Error('Failed to get user metadata');
+              }
+
+              // throw an error if the user is banned. Can't return a status, because the createNewSession doesn't allow for returning a status.
+              if (result.banned) {
+                throw new Error('User banned');
+              }
+
+              return originalImplementation.createNewSession(input);
+            },
+
+            getSession: async (input) => {
```

Comment:
We should probably check this in refreshSession - in that case, we can check in the DB, so bans will not go away for existing sessions on a server restart.